wgpu-core/src/resource.rs

```diff
@@ -1005,6 +1027,29 @@ impl Global {
 
         hal_surface_callback(hal_surface)
     }
+
+    /// # Safety
+    ///
+    /// - The raw command encoder handle must not be manually destroyed
+    pub unsafe fn command_encoder_as_hal_mut<
```

Comment:
the side effect of this requires it internally to start wgpu-core level recording if it hasn't started already should be documented both here and on the wgpu level

Reply:
Added.